contracts/contracts/echidna/EchidnaTestSupply.sol

```diff
@@ -0,0 +1,146 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "./EchidnaDebug.sol";
+import "./EchidnaTestTransfer.sol";
+
+import { StableMath } from "../utils/StableMath.sol";
+
+contract EchidnaTestSupply is EchidnaTestTransfer {
+    using StableMath for uint256;
+
+    uint256 prevRebasingCreditsPerToken = type(uint256).max;
+
+    // After a `changeSupply`, the total supply should exactly match the target
+    // total supply. (This is needed to ensure successive rebases are correct).
+    //
+    // testChangeSupply(uint256): failed!💥
+    //   Call sequence:
+    //       testChangeSupply(1044505275072865171609)
+    //
+    //   Event sequence:
+    //       TotalSupplyUpdatedHighres(1044505275072865171610, 1000000000000000000000000, 957391048054055578595)
+    //
+    function testChangeSupply(uint256 supply)
+        public
+        hasKnownIssue
+        hasKnownIssueWithinLimits
+    {
+        hevm.prank(ADDRESS_VAULT);
+        ousd.changeSupply(supply);
+
+        assert(ousd.totalSupply() == supply);
+    }
+
+    // The total supply must not be less than the sum of account balances. (The difference will go into future rebases)
+    //
+    // testTotalSupplyLessThanTotalBalance(): failed!💥
+    //   Call sequence:
+    //     mint(0,1)
+    //     changeSupply(1)
+    //     optOut(64)
+    //     transfer(0,64,1)
+    //     testTotalSupplyLessThanTotalBalance()
+    //
+    //   Event sequence:
+    //     Debug(«totalSupply», 1000000000000000001000001)
+    //     Debug(«totalBalance», 1000000000000000001000002)
+    //
+    function testTotalSupplyLessThanTotalBalance()
+        public
+        hasKnownIssue
+        hasKnownIssueWithinLimits
+    {
+        uint256 totalSupply = ousd.totalSupply();
+        uint256 totalBalance = getTotalBalance();
+
+        Debugger.log("totalSupply", totalSupply);
+        Debugger.log("totalBalance", totalBalance);
+
+        assert(totalSupply >= totalBalance);
```

Comment:
would be cool to test the limit in the other direction as well. To see we are within some acceptable range